torch/csrc/cudnn/AffineGridGenerator.cpp

```diff
@@ -0,0 +1,75 @@
+#include "AffineGridGenerator.h"
+
+#include "Descriptors.h"
+#include "Types.h"
+
+
+namespace torch { namespace cudnn {
+
+namespace {
+
+void setSamplerDescriptor(SpatialTransformerDescriptor& desc,
+			  cudnnDataType_t dataType,
+			  int N, int C, int H, int W)
+{
+  int inputSize[4] = {N, C, H, W};
+  desc.set(dataType, 4, inputSize);
+}
+
+void* tensorPointer(cudnnDataType_t dataType, THVoidTensor* tensor)
```

Comment:
It is also duplicated in BatchNorm.cpp, and Conv.cpp has a slightly different flavor of the same function.